### PR TITLE
http: use the request context

### PR DESCRIPTION
**Pull request by Stebalien**
CloseNotifier has been deprecated for a while.

Also, ditch the "environment" context. We don't actually _need_ this.

### CODE REVIEW

File: executor.go [hunk 1]
```diff
@@ -8,12 +8,8 @@ type Executor interface {
...
 }
 
-// Environment is the environment passed to commands. The only required method
-// is Context.
-type Environment interface {
-	// Context returns the environment's context.
-	Context() context.Context
-}
+// Environment is the environment passed to commands.
+type Environment interface{}
```
Comment (Stebalien) [inline executor.go]: (doesn't break anything in either go-filecoin or go-ipfs)

Comment (djdv) [reply executor.go]: Should `MakeEnvironment` be removed with this?

Comment (Stebalien) [reply executor.go]: We still need the environment (we type assert it to something more useful).

Really, we should consider stashing everything we'd _usually_ put in the environment in the _context_ (which is, IMO, where it should go) but we can handle that later.

File: http/parse.go [hunk 2]
```diff
@@ -1,7 +1,8 @@
 package http
 
 import (
-	"context"
+	"crypto/rand"
```
Comment (magik6k) [inline http/parse.go]: May want to use 'math/rand', it's 10-30x faster

**Pull request merged**